volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,1005 @@
+        """
+        Consistent way to get the filename regardless of platform
+
+        Args:
+            str: the file path from `filepath_for_address`
+
+        Returns:
+            str: the bsae file name of the full path
+        """
+        return ntpath.basename(filepath)
+
+    @staticmethod
+    def addresses_for_process_symbols(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        layer_name: str,
+        symbol_table_name: str,
+        symbols: filter_modules_type,
+    ) -> found_symbols_type:
+        """
+        Used to easily resolve the  addresses of names inside of modules.
+
+        See the usage of this function for system call resolution in  unhooked_system_calls.py
+        for an easy to understand example.
+
+        Args:
+            symbols: The dictionary of symbols requested by the caller
+
+        Returns:
+            found_symbols_type: The dictionary of symbols that were resolved
+        """
+        collected_modules = PESymbols.get_process_modules(
+            context, layer_name, symbol_table_name, symbols
+        )
+
+        found_symbols = PESymbols.find_symbols(
+            context, config_path, symbols, collected_modules
+        )
+
+        for mod_name, unresolved_symbols in symbols.items():
+            for symbol in unresolved_symbols:
+                vollog.debug(f"Unable to resolve symbol {symbol} in module {mod_name}")
+
+        return found_symbols
+
+    @staticmethod
+    def path_and_symbol_for_address(
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        collected_modules: collected_modules_type,
+        ranges: ranges_type,
+        address: int,
+    ) -> Tuple[str, str]:
+        """
+        Method for plugins to determine the file path and symbol name for a given address
+
+        See debugregisters.py for an example of how this function is used along with get_vads_for_process_cache
+        for resolving symbols in processes.
+
+        Args:
+            collected_modules: return value from `get_kernel_modules` or `get_process_modules`
+            ranges: the memory ranges to examine in this layer.
+            address: address to resolve to its symbol name
+        Returns:
+            Tuple[str|renderers.NotApplicableValue|renderers.NotAvailableValue, str|renderers.NotApplicableValue|renderers.NotAvailableValue]
+        """
+
+        if not address:
+            return renderers.NotApplicableValue(), renderers.NotApplicableValue()
+
+        filepath = PESymbols.filepath_for_address(ranges, address)
+
+        if not filepath:
+            return renderers.NotAvailableValue(), renderers.NotAvailableValue()
+
+        filename = PESymbols.filename_for_path(filepath).lower()
+
+        # setup to resolve the address
+        filter_module: filter_modules_type = {
+            filename: {wanted_addresses_identifier: [address]}
+        }
+
+        found_symbols = PESymbols.find_symbols(
```

Comment:
This doesn't look like it matches the function output anymore?

Reply:
Sorry, I mean slightly further down where we return found_symbols[filename][0][1]

Reply:
Sorry, find_symbols works exactly how it had, so this output will be right, I just thought it would returning a generator that then would provide a tuple.